### PR TITLE
Add new status PURGE_IN_PROGRESS and change purge process steps order

### PR DESCRIPTION
**Pull request by adidanes**
# Pull Request description
When undeploying a Deployment with purge option, if purging takes long time to be processed, we want that user be informed that Deployment status in puring instead of having the Deployment disapear (undeployed status).

## Description of the change
Add new DeploymentStatus corresponding to purging process is defined : UNDEPLOYMENT_PURGING.
Make work/deployment/<MyDeployment> directory remove step after other time consuming steps.

### What I did
When runPuge starts executing, set Deployment status to UNDEPLOYMENT_PURGING.
Move  work/deployment/<MyDeployment> directory remove step after removing from Consul KV store the MyDeployment's tasks, events and logs, which maybe long for some big applications.

### How I did it
Updated deployments/structs.go to add new purging status
Update runPurge in tasks/workflow/worker.go to set new purging status and change purging steps order.

## Applicable Issues
https://github.com/ystia/yorc/issues/504

### CODE REVIEW

File: tasks/workflow/worker.go [hunk 1]
```diff
@@ -592,11 +592,11 @@ func (w *worker) runUndeploy(ctx context.Context, t *taskExecution) error {
 }
 
 func (w *worker) runPurge(ctx context.Context, t *taskExecution) error {
+	// Set status to UNDEPLOYMENT_PURGING
+	deployments.SetDeploymentStatus(ctx, w.consulClient.KV(), t.targetID, deployments.UNDEPLOYMENT_PURGING)
```
Comment (loicalbertin) [inline tasks/workflow/worker.go]: Should we use `checkAndSetTaskStatus` instead?
@stebenoist WDYT?

**Comment by loicalbertin** [comment]
FYI build fails because Go 1.13 has just been released and some of our dependencies seems to not be valid anymore. A should be able to update it.

File: tasks/workflow/worker.go (hunk 1 above)
Comment (adidanes) [reply tasks/workflow/worker.go]: @loicalbertin, you mean that the new status should be a TaskStatus and not a DeploymentStatus ?

Comment (stebenoist) [reply tasks/workflow/worker.go]: IMO it's a deployment status so we should use SetDeploymentStatus.

File: .travis.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 language: go
 go:
-  - stable
```
Comment (loicalbertin) [inline .travis.yml]: Should go back to stable

File: tasks/workflow/worker.go (hunk 1 above)
Comment (loicalbertin) [reply tasks/workflow/worker.go]: Oups I was confused by those names... 
This is good to me.

File: .travis.yml [hunk 3]
```diff
@@ -1,6 +1,6 @@
...
+  - 1.12.x
```
Comment (loicalbertin) [inline .travis.yml]:
```suggestion
  - stable
```

File: tasks/workflow/worker_test.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+	// One log with value containing "content":"Status for deployment \"Test-Env\" changed to \"purged\"
+	kvps, _, err = kv.List(consulutil.LogsPrefix+"/"+deploymentID, nil)
+	require.True(t, len(kvps) == 1)
+	// One purge
+	kvps, _, err = kv.List(consulutil.PurgedDeploymentKVPrefix+"/"+deploymentID, nil)
+	require.True(t, len(kvps) == 1)
+}
+
+// Construct key/value to initialise KV before running test
+func testData(deploymentId string) map[string][]byte {
```
Comment (loicalbertin) [inline tasks/workflow/worker_test.go]: Should be deploymentID
```suggestion
func testData(deploymentID string) map[string][]byte {
```

File: tasks/workflow/worker_test.go [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+	return map[string][]byte{
+		// Add Test deployment
+		consulutil.DeploymentKVPrefix + "/" + deploymentId + "/status": []byte(deploymentId),
```
Comment (loicalbertin) [inline tasks/workflow/worker_test.go]:
```suggestion
		consulutil.DeploymentKVPrefix + "/" + deploymentID + "/status": []byte(deploymentID),
```

File: tasks/workflow/worker_test.go [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+		// deploy task
+		consulutil.TasksPrefix + "/t1/targetId": []byte(deploymentId),
```
Comment (loicalbertin) [inline tasks/workflow/worker_test.go]:
```suggestion
		consulutil.TasksPrefix + "/t1/targetId": []byte(deploymentID),
```

File: tasks/workflow/worker_test.go [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+		consulutil.TasksPrefix + "/t1/type":     []byte("0"),
+		// undeploy task
+		consulutil.TasksPrefix + "/t2/targetId": []byte(deploymentId),
```
Comment (loicalbertin) [inline tasks/workflow/worker_test.go]:
```suggestion
		consulutil.TasksPrefix + "/t2/targetId": []byte(deploymentID),
```

File: tasks/workflow/worker_test.go [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+		consulutil.TasksPrefix + "/t2/type":     []byte("1"),
+		// purge task
+		consulutil.TasksPrefix + "/t3/targetId": []byte(deploymentId),
```
Comment (loicalbertin) [inline tasks/workflow/worker_test.go]:
```suggestion
		consulutil.TasksPrefix + "/t3/targetId": []byte(deploymentID),
```

File: tasks/workflow/worker_test.go [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+		consulutil.TasksPrefix + "/t3/type":     []byte("4"),
+		// some events
+		// event should have "deploymentId":"Test-Env" and "type":"anyType but not purge"
+		consulutil.EventsPrefix + "/" + deploymentId + "/e1": []byte("aaaa"),
```
Comment (loicalbertin) [inline tasks/workflow/worker_test.go]:
```suggestion
		consulutil.EventsPrefix + "/" + deploymentID + "/e1": []byte("aaaa"),
```

File: tasks/workflow/worker_test.go [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+		consulutil.EventsPrefix + "/" + deploymentId + "/e2": []byte("bbbb"),
```
Comment (loicalbertin) [inline tasks/workflow/worker_test.go]:
```suggestion
		consulutil.EventsPrefix + "/" + deploymentID + "/e2": []byte("bbbb"),
```

File: tasks/workflow/worker_test.go [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+		// some logs
+		consulutil.LogsPrefix + "/" + deploymentId + "/l1":   []byte("xxxx"),
```
Comment (loicalbertin) [inline tasks/workflow/worker_test.go]:
```suggestion
		consulutil.LogsPrefix + "/" + deploymentID + "/l1":   []byte("xxxx"),
```

File: tasks/workflow/worker_test.go [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+		consulutil.EventsPrefix + "/" + deploymentId + "/l2": []byte("yyyy"),
```
Comment (loicalbertin) [inline tasks/workflow/worker_test.go]:
```suggestion
		consulutil.EventsPrefix + "/" + deploymentID + "/l2": []byte("yyyy"),
```

**Pull request merged**